blockchain/ethereum.go

```diff
@@ -335,18 +336,21 @@ func attemptReturn(e *EthereumClient, fromKey *ecdsa.PrivateKey, attemptCount in
 		return nil, err
 	}
 	totalGasCost := gasEstimations.TotalGasCost
-	addedBuffer := 1000000000 * attemptCount
-	totalGasCost.Add(totalGasCost, big.NewInt(int64(addedBuffer))) // Add 1 Gewi of buffer per attempt
-	balance.Sub(balance, totalGasCost)
+	balanceGasDelta := big.NewInt(0).Sub(balance, totalGasCost)
 
-	tx, err := e.NewTx(fromKey, nonce, to, balance, gasEstimations)
+	if balanceGasDelta.Cmp(big.NewInt(0)) <= 1 { // Try with 0.5 gwei if we have no or negative margin. Might as well
```

Comment:
@kalverra Currently running into an issue where the refund is always .5gwei. This check is always going to be true. Did you mean `balanceGasDelta.Cmp(big.NewInt(0)) <= -1` to check if `balanceGasDelta` is negative?